extensions/git/src/git.ts

```diff
@@ -2087,6 +2088,10 @@ export class Repository {
 			args.push('--unshallow');
 		}
 
+		if (options.autoStash) {
```

Comment:
We should not rely on the fact that the `autoStash` option would only be set if the version of git supports it. We should add a version check here. You can take a look at other places in the same file where we do that.